integration/container/health_test.go

```diff
@@ -93,6 +94,41 @@ while true; do sleep 1; done
 	poll.WaitOn(t, pollForHealthStatus(ctxPoll, client, id, "healthy"), poll.WithDelay(100*time.Millisecond))
 }
 
+// TestHealthCheckProcessKilled verifies that health-checks exec get killed on time-out.
+func TestHealthCheckProcessKilled(t *testing.T) {
+	skip.If(t, testEnv.OSType == "windows", "FIXME")
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(ctx, t, client, func(c *container.TestContainerConfig) {
+		c.Config.Healthcheck = &containertypes.HealthConfig{
+			Test:     []string{"CMD-SHELL", "sh -c 'trap \"echo SIGTERM\" 15; while true; do true; done;'"},
```

Comment:
I'm not so sure it's that `sleep` is blocking signals. I suspect that what's actually happening is busybox `ash` waits for the `sleep` subprocess to exit before executing the trap, [matching the behaviour of `bash`](https://www.gnu.org/software/bash/manual/html_node/Signals.html#:~:text=If%20Bash%20is,trap%20is%20executed.), and the daemon is only sending the signals to the topmost shell process so the `sleep` subprocess is never sent a signal.

Maybe something like this would work?
```sh
trap "echo SIGTERM" 15
sleep 30 &
pid=$!
while jobs %% >/dev/null; do wait $pid; done
```
I tested this interactively in `library/busybox:latest` and it seems to work, though the background job leaks.

Although, if it really is the case that the trap is not executing because the shell is waiting for the `sleep` command to complete, running `usleep` in a loop would effectively "poll" for the trapped signals with an upper-bound latency of the duration of the `usleep` command. No fancy stuff with background jobs required!
```suggestion
			Test:     []string{"CMD-SHELL", "sh -c 'trap \"echo SIGTERM\" 15; while true; do usleep 50000; done;'"},
```